### PR TITLE
build: officially deprecate waf

### PR DESCRIPTION
**Pull request by Dudemanguy**
While the waf build has served us well for many years, it's time to officially consider it deprecated. The meson build was added fully with the intention to eventually replace waf and its current state is more than good enough to do that. Let's start the deprecation period now to give users a heads up to switch before we remove waf for good.

Please let me know if I missed something somewhere.

P.S. if somebody could update `compile-windows.md` with meson instructions that would be great.

### CODE REVIEW

**Review by philipl — APPROVED** [review]

**Comment by Akemi** [comment]
when merged we should also close all issues labeled as core:waf (only).

**Comment by garoto** [comment]
I mean, like the build system was the most pressing issue with the player currently...

**Comment by cloud11665** [comment]
End of an era...

**Comment by Dudemanguy** [comment]
>I mean, like the build system was the most pressing issue with the player currently...

For users it may not matter so much, but for developers the build system is pretty important. mpv's waf system is very custom and incredibly opaque, and it's "documentation" (lol) looks like it was written by a GPT bot. For any potential contributors, I guarantee that my meson mess is easier to understand than the stuff that waf does under the hood.

**Comment by maxdunbar** [comment]
> > I mean, like the build system was the most pressing issue with the player currently...
> 
> For users it may not matter so much, but for developers the build system is pretty important. mpv's waf system is very custom and incredibly opaque, and its "documentation" (lol) looks like it was written by a GPT bot. For any potential contributors, I guarantee that my meson mess is easier to understand than the stuff that waf does under the hood.

I can attest to this. When building for the first time, I had to wrestle with waf. Whereas meson "just worked'!

**Review by LaserEyess — COMMENTED** [review]
I think https://github.com/mpv-player/mpv/blob/master/DOCS/compile-windows.md should be updated as well, it's very waf-heavy, so the instructions should switch to being meson heavy. I took a stab at that here: https://github.com/mpv-player/mpv/blob/3fd998e7760bfc78c793cf8a9e9b454edcb200d7/DOCS/compile-windows.md

**Comment by Dudemanguy** [comment]
Now includes a commit to update `compile-windows.md` to meson instructions.

**Review by LaserEyess — APPROVED** [review]

**Comment by Hrxn** [comment]
Just noticed, `compile-windows.md` doesn't seem to include any instructions on how to build statically linked binaries, or is that not trivially possible on Windows?
Or what am I missing here?

**Comment by Dudemanguy** [comment]
You can use the `prefer-static` option in meson. No clue why this isn't in the guide (static linking wasn't mentioned in the waf version either).

File: DOCS/compile-windows.md [hunk 1]
```diff
@@ -4,28 +4,43 @@ Compiling for Windows
...
 
 Cross-compilation
 =================
 
-When cross-compiling, you have to run mpv's configure with these arguments:
-
-```bash
-DEST_OS=win32 TARGET=i686-w64-mingw32 ./waf configure
+When cross-compiling, it is recommended to use a meson crossfile to setup
+the cross compiling environment. A minimal eaxmple is included below:
```
Comment (eli-schwartz) [inline DOCS/compile-windows.md:20]:
```suggestion
the cross compiling environment. A minimal example is included below:
```

**Comment by eli-schwartz** [comment]
> I mean, like the build system was the most pressing issue with the player currently...

If it's not the most pressing issue then why do you care which one is used? This PR is just ordinary non-user-affecting administrivium.

File: DOCS/compile-windows.md (hunk 1 above)
Comment (Dudemanguy) [reply DOCS/compile-windows.md:20]: woops

**Comment by FarisR99** [comment]
Following the Windows compilation instructions using MSYS2, mpv.exe is built but mpv.com is nowhere to be found. Not sure if it's just my setup. This happens both with `-Dlibmpv=true` and without.

**Pull request merged**